pyproject.toml

```diff
@@ -1,34 +1,184 @@
-
 [build-system]
 requires = [
-  # sync with setup.py until we discard non-pep-517/518
-  "setuptools",
-  "setuptools-scm",
-  "wheel",
-  "build",
+    "hatchling>=1.8.0",
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "semver"
+description = "Python helper for Semantic Versioning (https://semver.org)"
+readme = "README.rst"
+requires-python = ">=3.7"
+authors = [
+    { name = "Kostiantyn Rybnikov", email = "k-bx@k-bx.com" },
+]
+maintainers = [
+    { name = "Sebastien Celles", email = "s.celles@gmail.com" },
+    { name = "Tom Schraitle" },
+]
+keywords = [
+    "python",
+    "version",
+    "semver",
+    "versioning",
+    "version",
+    "semantic-versioning",
+    "release",
+    "semver-format",
+    "semver-tag",
+    "semver-release",
+    "semver-cli",
+]
+classifiers = [
+    # I think python-semver is more "Console" than "Web Environment"
+    #"Environment :: Web Environment",
+    "Environment :: Console",
+    "Development Status :: 5 - Production/Stable",
+    "Topic :: Software Development",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = []
+dynamic = [
+    "version",
+]
+
+[project.urls]
+Homepage = "https://github.com/python-semver/python-semver"
+Changelog = "https://python-semver.readthedocs.io/en/latest/changelog.html"
+Documentation = "https://python-semver.rtfd.io"
+Releases = "https://github.com/python-semver/python-semver/releases"
+Issues = "https://github.com/python-semver/python-semver/issues"
+
+[project.license]
+file = "LICENSE.txt"
+
+[project.scripts]
+pysemver = "semver.cli:main"
+
+[tool.hatch.version]
+path = "src/semver/__about__.py"
+
+[tool.hatch.build]
+include = [
+    "src/semver/*.py",
+    "src/semver/py.typed",
+]
+
+[tool.hatch.envs.default]
+dependencies = [
+    "towncrier",
+    "wheel",
+]
+
+[tool.hatch.envs.style]
+dependencies = [
+    "black",
+    "isort",
+    "flake8",
+    "pycodestyle",
+]
+
+[tool.hatch.envs.style.scripts]
+fmt = [
+    "black .",
+    "isort .",
+]
+lint = [
+    "flake8 --exit-zero",
+    "pycodestyle",
 ]
-build-backend = "setuptools.build_meta"
 
+[tool.hatch.envs.docs]
+dependencies = [
+    "pdoc3"
+]
+
+[tool.hatch.envs.docs.scripts]
+build = "pdoc --html --output-dir api-docs src/semver --force"
+serve = "pdoc --http : python-semver"
+
+[tool.hatch.envs.test]
+dependencies = [
+    "pytest-cov",
+    "tox",
+]
+
+[tool.hatch.envs.test.scripts]
+cov = "pytest -vx"
+no-cov = "cov --no-cov"
+tox_test = "tox"
```

Comment:
I suspect something is missing here. When I run tox (or pytest), I can run it either tests against all defined Python version or a specific version. Running `tox` alone would always run _all_ Python versions. That's a time consuming task and you maybe want to restrict it to run against Python 3.7 only.

Reply:
umm, sure. I think it would be convenient to use the `test` environment for testing only with Python 3.7, and have another test environment, e.g. `testall` with the complete matrix of Python versions.

For running tox with all its regular tests I will prepare a `hatch run tox:test` script.